addons/api/tests/unit/serializers/host-test.js

```diff
@@ -4,19 +4,128 @@ import { setupTest } from 'ember-qunit';
 module('Unit | Serializer | host', function (hooks) {
   setupTest(hooks);
 
-  test('it exists', function (assert) {
-    let store = this.owner.lookup('service:store');
-    let serializer = store.serializerFor('host');
-
-    assert.ok(serializer);
+  test('it serializes host with type static as expected', function (assert) {
+    assert.expect(1);
+    const store = this.owner.lookup('service:store');
+    const record = store.createRecord('host', {
+      name: 'Host 1',
+      compositeType: 'static',
+      description: 'Description',
+      host_catalog_id: '123',
+      version: 1,
+      address: '188e:68a9:b342:c05c:2595:2f46:499c:759f',
+    });
+    assert.deepEqual(record.serialize(), {
+      name: 'Host 1',
+      description: 'Description',
+      type: 'static',
+      host_catalog_id: '123',
+      version: 1,
+      attributes: {
+        address: '188e:68a9:b342:c05c:2595:2f46:499c:759f',
+      },
+    });
   });
 
-  test('it serializes records', function (assert) {
-    let store = this.owner.lookup('service:store');
-    let record = store.createRecord('host', {});
+  test('it serializes host with type plugin as expected', function (assert) {
+    assert.expect(1);
+    const store = this.owner.lookup('service:store');
+    const recordAws = store.createRecord('host', {
+      name: 'Host 2',
+      description: 'Description',
+      host_catalog_id: '123',
+      compositeType: 'aws',
+      address: '188e:68a9:b342:c05c:2595:2f46:499c:759f',
+    });
+
+    assert.deepEqual(recordAws.serialize(), {
+      name: 'Host 2',
+      description: 'Description',
+      host_catalog_id: '123',
+      type: 'plugin',
+      attributes: {
+        address: '188e:68a9:b342:c05c:2595:2f46:499c:759f',
+      },
+    });
+  });
 
-    let serializedRecord = record.serialize();
+  test('it normalizes type static record', function (assert) {
+    assert.expect(1);
+    const store = this.owner.lookup('service:store');
+    const serializer = store.serializerFor('host');
+    const host = store.createRecord('host').constructor;
+    const payload = {
+      id: '1',
+      name: 'host test',
+      type: 'static',
+      ip_addresses: ['10.0.0.1', '10.0.0.2', '10.0.0.3'],
```

Comment:
Technically, `ip_addresses` and `dns_names` do not apply to `static` hosts.  Only `attributes.address` applies to `static`.